### PR TITLE
add jenkins file

### PR DESCRIPTION
**Pull request by jhaugh42**
so this repo can be built and published via CI

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+    stage('publish') {
+      when {
+        branch 'master'
+      }
+      steps {
+        script {
+          withVaultCredentials([
+            [path: '/secret/content-engineering/global/artifactory/content-engineering@gannett.com', keys: ['username': 'ARTIFACTORY_USERNAME', 'password': 'ARTIFACTORY_PASSWORD']],
+            [path: "/secret/content-engineering/global/tyk/paas-api", keys: ['TYK_API_KEY': 'TYK_API_KEY']],
```
Comment (ibeyene) [inline Jenkinsfile]: we are no longer using TYK_API, change it to apigee. The path is ```/secret/content-engineering/global/apigee/paas-api``` 

keys: ```['API_KEY': 'API_KEY']],```

Comment (jhaugh42) [reply Jenkinsfile]: Good call, will update.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            if (repo_package_version == node_package.version) {
+             env.FAILURE_MESSAGE = "<${env.BUILD_URL}|#${env.BUILD_TAG}> - [${node_package.name}] node_package.version:${node_package.version} already exists in npm repo as repo_package_version:${repo_package_version}. update the version number in package.json"
+             error env.FAILURE_MESSAGE
+            }
+            sh 'git add package.json package-lock.json'
+            sh "git -c user.name='${node_package.author.name}' -c user.email='${node_package.author.name}@gannett.com' commit -m '[ci auto incrementing version] to ${node_package.version}'"
+            sh "git push https://${env.GIT_PASSWORD}:${env.GIT_PASSWORD}@github.com/GannettDigital/${env.GIT_REPO_NAME} ${env.BRANCH_NAME}"
+            def pkg = sh(script: 'npm pack .', returnStdout: true).trim()
+            archiveArtifacts artifacts: pkg
+            paasTag.create apiKey: env.TYK_API_KEY,
```
Comment (akemner) [inline Jenkinsfile:59]: `API_KEY` not `TYK_ API_KEY`

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,110 @@
+pipeline {
+  agent {
+    dockerfile true
+  }
+  environment {
+    def NPM_CONFIG_CACHE = '/tmp/npm'
```
Comment (akemner) [inline Jenkinsfile:6]: missing `ARTIFACTORY_REPO = 'cs-node'`

**Pull request merged**